File: WebContent/WEB-INF/jython/category_query.py
# -*- coding: utf-8 -*-
from cn.edustar.jitar.data import BaseQuery

# 分类查询
class CategoryQuery (BaseQuery):
    def __init__(self, selectFields):
      BaseQuery.__init__(self, selectFields)
      #分类的父分类, 缺省 = null
      self.parentId = None
      
      #分类的类型　缺省是文章分类
      self.itemType = 'default'
      
      
    def initFromEntities(self, qctx):
      qctx.addEntity("Category" , "cat" , "")
    
    def resolveEntity(self, qctx, entity):
        BaseQuery.resolveEntity(self, qctx, entity)
      
    # 提供 where 条件.
    def applyWhereCondition(self,qctx):
      if self.parentId == None:
        qctx.addAndWhere("cat.parentId = NULL")
      elif self.parentId == 0:
        qctx.addAndWhere("cat.parentId = NULL")
      else:
        qctx.addAndWhere("cat.parentId = :parentId")
        qctx.setInteger("parentId", self.parentId)
        
      if self.itemType != None:
        qctx.addAndWhere("cat.itemType = :itemType")
        qctx.setString("itemType", self.itemType)
        
    # 提供排序 order 条件.
    def applyOrderCondition(self, qctx):
      # 排序方式, 0 - commentId DESC, 1 - createDate
        qctx.addOrder("cat.orderNum ASC") 
      

File: src/cn/edustar/jitar/service/CategoryQuery.java
package cn.edustar.jitar.service;

import cn.edustar.jitar.data.BaseQuery;
import cn.edustar.jitar.data.QueryContext;

public class CategoryQuery  extends BaseQuery {
    //#分类的父分类, 缺省 = null
    public Integer parentId = null;
    
    //#分类的类型　缺省是文章分类
    public String itemType = "default";
    
	public CategoryQuery(String selectFields) {
		super(selectFields);
	}
    @Override
    public void initFromEntities(QueryContext qctx) {
    	qctx.addEntity("Category" , "cat" , "");
    }
    
    public void resolveEntity(QueryContext qctx, String entity) {
    	super.resolveEntity(qctx, entity);
    }
    
    public void applyWhereCondition(QueryContext qctx) {
        if (this.parentId == null){
            qctx.addAndWhere("cat.parentId = NULL");
        }else if(this.parentId == 0){
            qctx.addAndWhere("cat.parentId = NULL");
        }else{
            qctx.addAndWhere("cat.parentId = :parentId");
            qctx.setInteger("parentId", this.parentId);
        }
        if (this.itemType != null){
            qctx.addAndWhere("cat.itemType = :itemType");
            qctx.setString("itemType", this.itemType);
        }
    	
    }
    
    public void applyOrderCondition(QueryContext qctx) {
    	qctx.addOrder("cat.orderNum ASC");
    }
}
